### ISSUE TITLE
Upload SciPy tutorial dataset to OpenML

### ISSUE THREAD

**Issue by hildeweerts** [fairlearn/fairlearn#1064]
As discussed during last week's community call, we need to upload the preprocessed dataset that was used for the SciPy tutorial to OpenML, so it can be easily fetched in examples.

- [ ] check licensing
- [ ] write a short description of the pre-processed dataset
- [ ] upload dataset to OpenML

I would be happy to upload the dataset, but I believe @MiroDudik had possible concerns re. licensing. Would crediting the original authors and/or UCI repository suffice? Are there any additional concerns I need to take into account before uploading the data?

Tagging @LeJit @MiroDudik

**Comment by LeJit** [comment]
I was looking into this last week. When you submit a dataset, there's an **Additional Information** section where we can link to the *Original Data URL*, list *Acknowledgements, contributors*, and *Paper/preprint*. 

I think we could credit the original authors in this section.

**Comment by hildeweerts** [comment]
Yeah, OpenML gives the opportunity to credit authors! 

Depending on the specific license that might not be enough though. The original dataset was from UCI, right? from the UCI website:
"Datasets will be licensed under a Creative Commons Attribution 4.0 International license (CC BY 4.0) which allows for the sharing and adaptation of the datasets for any purpose, provided that the appropriate credit is given (see Citation Policy)."

I couldn't find said citation policy but it seems we're good as long as we credit the authors!

**Comment by LeJit** [comment]
<img width="588" alt="OpenML" src="https://user-images.githubusercontent.com/1124464/162041420-99ebb95d-ee60-48c5-a477-34e429a51e57.png">
<img width="584" alt="OpenML_part2" src="https://user-images.githubusercontent.com/1124464/162041422-6f6d3146-0a8d-4f3b-b409-fd0062f13d70.png">

Here's what I have right now. Do we want to list ourselves as the *authors*, or should we put the original authors in this section?

Also, here's the full *description* I'm using:

> This dataset is a processed version of the "U.S. Diabetes 130-Hospital". Contributors to the Fairlearn library created this dataset for their tutorial "Fairness in AI Systems: From social context to practice using Fairlearn", presented at SciPy 2021.

> The original dataset was collected by Beata Strack, Jonathan P. DeShazo, Chris Gennings, Juan L. Olmo, Sebastian Ventura, Krzysztof J. Cios, and John N. Clore in 2014.

**Comment by hildeweerts** [comment]
I agree with Roman that the credit should go to the original authors. We should definitely add to the description that and how we edited the original dataset though. 

In my view the description should describe the contents of the dataset, such that people do not necessarily need to visit other websites to understand what it's about. Perhaps we can swap the two paragraphs in the description: start with the original dataset (was collected by X and Y etc) and give a short description of the prediction task and how the dataset was collected (possibly copied from UCI). Then we can note that we pre-processed it, including a short summary of how it is different from the original dataset.

Also note that the name will be the ID of the dataset so short and sweet is typically better.

**Comment by LeJit** [comment]
Updated description:

> The "Diabetes 130-Hospitals" dataset represents 10 years of clinical care at 130 U.S. hospitals and delivery networks, collected from 1999 to 2008. Each record represents the hospital admission record for a patient diagnosed with diabetes whose stay lasted between one to fourteen days. The features describing each encounter include demographics, diagnoses, diabetic medications, number of visits in the year preceding the encounter, and payer information, as well as whether the patient was readmitted after release, and whether the readmission occurred within 30 days of the release.
> 
> The original "Diabetes 130-Hospitals" dataset was collected by Beata Strack, Jonathan P. DeShazo, Chris Gennings, Juan L. Olmo, Sebastian Ventura, Krzysztof J. Cios, and John N. Clore in 2014.
> 
> This version of the dataset was edited by the Fairlearn team for the SciPy 2021 tutorial "Fairness in AI Systems: From social context to practice using Fairlearn".  In this version, the target variable "readmitted" is binarized into whether the patient was re-admitted within thirty days. The full dataset pre-processing script can be found on GitHub: https://github.com/fairlearn/talks/blob/main/2021_scipy_tutorial/preprocess.py

@MiroDudik @hildeweerts What do you think?

**Comment by MiroDudik** [comment]
Looks good. I would just:
* add URL (pointing to UCI) of the original data set right after "2014."
* small edit:
  * "was edited by the Fairlearn team" -> "was derived from the original dataset by the Fairlearn team"

**Comment by hildeweerts** [comment]
Perfect, @LeJit!

**Comment by adrinjalali** [comment]
Is this done?

**Comment by hildeweerts** [comment]
There were some issues on the OpenML side (the data upload seems to be broken on the new website or something). But @LeJit was working on uploading the dataset through the Python API instead (which should still work).

**Comment by LeJit** [comment]
I managed to get it to work through the Python API. Here's the URL for the dataset: https://www.openml.org/d/43874

**Comment by adrinjalali** [comment]
Nice, then we can close this issue.
